docs/design-documents/platform/memory-model/ram_memory_model.md

```diff
@@ -0,0 +1,105 @@
+
+                +----------------------+ Stack Start (Last address of RAM)
+                | ISR stack            |
+                | Main Stack(No RTOS)  |
+                |          |           |
+                |          V           |
+                +----------------------+
+                |          ^           |
+                |          |           |
+                |        Heap          |
+                +----------------------+ HEAP Start
+                |         ZI           |
+                |(Idle, Timer and Main |
+                | stack is in ZI for   |
+                | RTOS)                |
+                +----------------------+
+                |                      |
+                +----------------------+ First address of RAM
+
+#### Drawbacks:
+1. Collisions between stack and heap are hard to detect and result in hardfault.
+1. Cannot check stack limit - In case of new ARM architecture stack limit registers are available to verify stack boundaries, but this feature cannot be used with dynamic stack size.
+1. Stack size unification cannot be achieved across various targets.
+1. GCC ARM: Memory allocator request memory at 4K boundary end of HEAP memory should be 4K aligned. Placing ISR stack (1K) after HEAP memory in case of RTOS, results in loss of 3K RAM memory
+1. Memory alloctors do not support HEAP split into multiple banks, hence with single region memory model HEAP is used only till end of first bank.
+
+### Proposed RAM memory model
+
+2-region memory model for heap and stack. Defined boundaries for ISR stack memory. Heap memory can be dynamic (starts at end of ZI and ends at last RAM address) or with fix boundaries in separate RAM bank.
+
+                +----------------------+ Heap Ends (Last address of RAM)
+                |          ^           |
+                |          |           |
+                |        Heap          |
+                +----------------------+ HEAP Start
+                |         ZI           |
+                |(Idle, Timer and Main |
+                | stack is in ZI for   |
+                | RTOS)                |
+                +----------------------+Stack Ends 
+                | ISR stack            |
+                | Main Stack(No RTOS)  |
+                |          |           |
+                |          V           |
+                +----------------------+Stack Start
+                |                      |
+                +----------------------+ First address of RAM
+
+#### Drawbacks:
+1. ISR Stack is not dynamic - This drawback is mainly for bare metal implementation (RTOS-less) where ISR and Main stack is same. With this limitation application writer should know if stack or heap will be usued more and tweaks the values accordingly.
+
+# Phases:
+This feature will be implemented in different phases as follow:
+
+Phase 1 (5.12 Release):
```

Comment:
so what phase is being implemented in this PR ? is it Phase 1 only ?

Reply:
Yes Phase 1 in this PR (5.12)